lib/ukvmem/Config.uk

```diff
@@ -71,6 +71,18 @@ config LIBUKVMEM_STACK_BASE
 	default 0x0000004000000000
 	depends on LIBUKVMEM_DEDICATED_STACK_BASE
 
+config LIBUKVMEM_STACK_GUARD_PAGES_TOP
+	int "Top guard pages count"
+	default 4
```

Comment:
It's based on intuition after noticing that 2 pages are not enough for a Wordpress based application.

Reply:
In fact the issue was related to on-stack array access. Since stacks are growing downwards and arrays upwards, you easily get outside of a single 4k page when accessing index 0 of a bigger array. On one hand I agree with you @michpappas and on the other hand this should be a protective feature. I found that there is a [POSIX interface](https://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_attr_getguardsize.html) for it, so it should be ultimately a per-thread property to make it right. I can imagine that the rational behind this coulb probably be that an application can estimate typical access distances and size the guard pages accordingly.
At this point, I would accept the 4 pages for user experience but longer term we need to change the integration so that it becomes configurable per thread -> I think this requires some API discussion first.